From d76be85d426227a45d17f0ad7da63f72f2d41015 Mon Sep 17 00:00:00 2001
From: "Shankarmurthy, Akshay" <akshay.s@ti.com>
Date: Mon, 1 Oct 2012 11:08:57 +0530
Subject: [PATCH] da830_wifi_evm: add an entry in boards.cfg for da830 wifi
 evm

This patch adds support in boards.cfg for da830_wifi_evm.
Since Ethernet switch of this board doesn't require I2C
enable so the execution of that is avoided by putting it
under conditional compilation.

Signed-off-by: Shankarmurthy, Akshay <akshay.s@ti.com>
---
 board/davinci/da8xxevm/da830evm.c |    4 ++++
 boards.cfg                        |    1 +
 2 files changed, 5 insertions(+)

diff --git a/board/davinci/da8xxevm/da830evm.c b/board/davinci/da8xxevm/da830evm.c
index c45c94b..aa87835 100644
--- a/board/davinci/da8xxevm/da830evm.c
+++ b/board/davinci/da8xxevm/da830evm.c
@@ -254,6 +254,7 @@ int board_nand_init(struct nand_chip *nand)
  */
 int board_eth_init(bd_t *bis)
 {
+#ifndef CONFIG_DA830_WIFI_EVM
 	u_int8_t mac_addr[6];
 	u_int8_t switch_start_cmd[2] = { 0x01, 0x23 };
 	struct eth_device *dev;
@@ -273,6 +274,7 @@ int board_eth_init(bd_t *bis)
 		printf("Ethernet switch start failed!\n");
 		return -1;
 	}
+#endif
 
 	/* finally, initialise the driver */
 	if (!davinci_emac_initialize()) {
@@ -280,12 +282,14 @@ int board_eth_init(bd_t *bis)
 		return -1;
 	}
 
+#ifndef CONFIG_DA830_WIFI_EVM
 	dev = eth_get_dev();
 
 	/* provide the resulting addr to the driver */
 	memcpy(dev->enetaddr, mac_addr, 6);
 	dev->write_hwaddr(dev);
 
+#endif
 	return 0;
 }
 #endif /* CONFIG_DRIVER_TI_EMAC */
diff --git a/boards.cfg b/boards.cfg
index a49ce89..c4bb84c 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -120,6 +120,7 @@ pm9263                       arm         arm926ejs   pm9263              ronetix
 pm9g45                       arm         arm926ejs   pm9g45              ronetix        at91        pm9g45:AT91SAM9G45
 cam_enc_4xx                  arm         arm926ejs   cam_enc_4xx         ait            davinci     cam_enc_4xx
 da830evm                     arm         arm926ejs   da8xxevm            davinci        davinci
+da830evm_wifi                arm         arm926ejs   da8xxevm            davinci        davinci	    da830evm:DA830_WIFI_EVM
 da850_am18xxevm              arm         arm926ejs   da8xxevm            davinci        davinci     da850evm:DA850_AM18X_EVM,MAC_ADDR_IN_EEPROM,SYS_I2C_EEPROM_ADDR_LEN=2,SYS_I2C_EEPROM_ADDR=0x50
 da850evm                     arm         arm926ejs   da8xxevm            davinci        davinci     da850evm:MAC_ADDR_IN_SPIFLASH
 da850evm_nand                arm         arm926ejs   da8xxevm            davinci        davinci     da850evm:MAC_ADDR_IN_SPIFLASH,USE_NAND
-- 
1.7.9.5

